### PR TITLE
[xen] Document the patches.

### PR DESCRIPTION
**Pull request by rossphilipson**
Used information from here:
https://openxt.atlassian.net/wiki/display/DC/Xen+4.6+Uprev+-+Patch+queue

Used information from git history.

Used information from my head.

This takes care of all the patches not covered by specific tickets.

OXT-610

Signed-off-by: Ross Philipson <philipsonr@ainfosec.com>

### CODE REVIEW

File: recipes-extended/xen/files/disable-xen-root-check.patch [hunk 1]
```diff
@@ -1,10 +1,12 @@
 ################################################################################
 SHORT DESCRIPTION: 
 ################################################################################
+Don't for XEN_ROOT path to be absolute.
```
Comment (jean-edouard) [inline recipes-extended/xen/files/disable-xen-root-check.patch]: Missing word?

File: recipes-extended/xen/files/hvm-cpuid-multicore.patch [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
 SHORT DESCRIPTION: 
 ################################################################################
+Two CPUID adjusting patches in one, see description.
 
 ################################################################################
 LONG DESCRIPTION: 
 ################################################################################
+This is actually two CPUID related patches in one. The part about the
+XciVMMXciVMM signature was used by our old custom Linux PV drivers to get them
+to load instead of standardd pvops ones. We keep this feature because it is
```
Comment (jean-edouard) [inline recipes-extended/xen/files/hvm-cpuid-multicore.patch]: Typo (standardd)

File: recipes-extended/xen/files/hvm-cpuid-multicore.patch [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+very useful for debugging. By enabling this CPUID you can for a pvops guest
```
Comment (jean-edouard) [inline recipes-extended/xen/files/hvm-cpuid-multicore.patch]: Missing word?

File: recipes-extended/xen/files/xen-translate.patch [hunk 4]
```diff
@@ -1,10 +1,15 @@
 ################################################################################
 SHORT DESCRIPTION: 
 ################################################################################
+Convert guest framebuffer GFNs to MFNs for Surfman gem_foreign support.
 
 ################################################################################
 LONG DESCRIPTION: 
 ################################################################################
+This is done so Surfman can manage framebuffer's MFN directly, as it used to
+registered those directly to the IGD using former IGFX plugin. This can probably
```
Comment (jean-edouard) [inline recipes-extended/xen/files/xen-translate.patch]: `s/registered/register/`?

**Comment by rossphilipson** [comment]
Fixed that stuffs, did a commit.

**Comment by jean-edouard** [comment]
LGTM, merging tomorrow.

**Pull request merged**